src/macaron/repo_finder/provenance_extractor.py

```diff
@@ -355,3 +358,354 @@ def check_if_repository_purl_and_url_match(url: str, repo_purl: PackageURL) -> b
+
+        Parameters
+        ----------
+        statement : InTotoV1Statement | InTotoV01Statement
+            The provenance statement from which to extract the build invocation
+            details. This statement contains the metadata about the build process
+            and its associated artifacts.
+
+        Returns
+        -------
+        tuple[str | None, str | None]
+            A tuple containing two elements:
+            - The first element is the build invocation entry point (e.g., workflow name), or None if not found.
+            - The second element is the invocation URL or identifier (e.g., job URL), or None if not found.
+
+        Raises
+        ------
+        NotImplementedError
+            If the method is called directly without being overridden in a subclass.
+        """
+
+
+class SLSAGithubGenericBuildDefinitionV01(ProvenanceBuildDefinition):
+    """Class representing the SLSA GitHub Generic Build Definition (v0.1).
+
+    This class implements the abstract methods defined in `ProvenanceBuildDefinition`
+    to extract build invocation details specific to the GitHub provenance generator's generic build type.
+    """
+
+    #: Determines the expected ``buildType`` field in the provenance predicate.
+    expected_build_type = "https://github.com/slsa-framework/slsa-github-generator/generic@v1"
+
+    def get_build_invocation(self, statement: InTotoV01Statement | InTotoV1Statement) -> tuple[str | None, str | None]:
+        """Retrieve the build invocation information from the given statement.
+
+        This method is intended to be implemented by subclasses to extract
+        specific invocation details from a provenance statement.
+
+        Parameters
+        ----------
+        statement : InTotoV1Statement | InTotoV01Statement
+            The provenance statement from which to extract the build invocation
+            details. This statement contains the metadata about the build process
+            and its associated artifacts.
+
+        Returns
+        -------
+        tuple[str | None, str | None]
+            A tuple containing two elements:
+            - The first element is the build invocation entry point (e.g., workflow name), or None if not found.
+            - The second element is the invocation URL or identifier (e.g., job URL), or None if not found.
+        """
+        if statement["predicate"] is None:
+            return None, None
+        gha_workflow = json_extract(statement["predicate"], ["invocation", "configSource", "entryPoint"], str)
+        gh_run_id = json_extract(statement["predicate"], ["invocation", "environment", "github_run_id"], str)
+        repo_uri = json_extract(statement["predicate"], ["invocation", "configSource", "uri"], str)
+        repo = None
+        if repo_uri:
+            repo = _clean_spdx(repo_uri)
+        if repo is None:
+            return gha_workflow, repo
+        invocation_url = f"{repo}/" f"actions/runs/{gh_run_id}"
```

Comment:
I wonder if it's intended to concatenate 2 formatted string? 
nit: we could use one single formatted string `f"{repo}/actions/runs/{gh_run_id}"`